src/muya/lib/parser/marked/inlineRules.js

```diff
@@ -37,7 +37,7 @@ const inline = {
   emoji: noop,
 
   // TODO: make math optional GH#740
-  math: /^\$([^$]{1}[\s\S]+?[^$]{1})\$(?!\$)/
+  math: /^\$([^$]*?[^\$\\])\$(?!\$)/
```

Comment:
Why do you match `0-unlimeted` and optional?

Reply:
Because there is a `[^\$\\]` after it, so it match `1~unlimeted` between `$` and `$`.